.github/workflows/build.yml

```diff
@@ -7,12 +7,62 @@ on:
     branches: [ "master" ]
 
 jobs:
-
-  build:
-
+  code-formatting:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v3
+    - name: Set up node
+      uses: actions/setup-node@v3
+    - name: Install npm packages
+      run: npm ci
+    - name: Run code formatting tests
+      run: ./tests/run-formatting-tests.sh
+    - name: Run Ruby formatting tests
+      run: ./tests/run-ruby-linting-tests.sh
+  unit-tests:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v3
+    - name: Build Docker backend
+      run: |
+        cp dev.env .env
+        docker-compose up --build -d
+        docker ps -a
+        docker-compose run backend rake db:create
+        docker-compose run backend rake db:migrate
+    - name: Display Docker logs on failure
+      if: ${{ failure() }}
+      run: docker-compose logs
+    - name: Run unit tests
+      run: ./tests/run-unit-tests.sh
+    - name: Display Docker logs on failure
+      if: ${{ failure() }}
+      run: docker-compose logs
+    - name: Clean up Docker
+      run: docker-compose down
+  production-build:
+    if: ${{ github.event.issue.push }}
+    needs: [code-formatting, unit-tests]
```

Comment:
Let's drop this. It's useful to know if the build is broken separately from whether code formats correctly or unit tests run.